File: apr/apr2/lc690.py
class Solution:
    def getImportance(self, employees: List['Employee'], id: int) -> int:
        map = {}
        for i in employees:
            map[i.id] = i
        def dfs(root):
            x = root.importance
            for i in root.subordinates:
                x += dfs(map[i])
            return x
        return dfs(map[id])

File: june/7june/lc690.java
class Solution {
    Map<Integer,Employee> gemployees = new HashMap<>();
    int traverse(Employee root){
        int c = 0;
        for(int e:root.subordinates){
            c += traverse(gemployees.get(e));
        }
        return root.importance + c;
    }
    public int getImportance(List<Employee> employees, int id) {
        for(Employee e:employees){
            gemployees.put(e.id,e);
        }
        return traverse(gemployees.get(id));
    }
}